systems/daint-mc.sh

```diff
@@ -41,6 +41,7 @@ ns_sockets=2
 ns_threads_per_socket=36
 
 run_with_mpi() {
-    echo ARB_NUM_THREADS=$ns_threads_per_socket srun -n $ns_sockets -c $ns_threads_per_socket "${@}"
-    ARB_NUM_THREADS=$ns_threads_per_socket srun -n $ns_sockets -c $ns_threads_per_socket "${@}" 
+    export ARB_NUM_THREADS=$ns_threads_per_socket
+    echo srun -n $ns_sockets -c $ns_threads_per_socket "${@}"
+    srun -n $ns_sockets -c $ns_threads_per_socket "${@}"
```

Comment:
The `-C` constraint should be added so that benchmarks and tests that are run without pre-allocation with `salloc` will not error out.

(Also applies to daint-gpu.sh above.)